crypto/src/traits.rs

```diff
@@ -172,7 +173,18 @@ pub trait KeyPair: Sized {
         ikm: &[u8],  // IKM (32 bytes)
         salt: &[u8], // Optional salt
         info: &[u8], // Optional domain
-    ) -> Result<Self, signature::Error>;
+    ) -> Result<Self, signature::Error> {
+        let hk = hkdf::Hkdf::<Sha3_256>::new(Some(salt), ikm);
+        const SHA3_OUTPUT_SIZE: usize = <Sha3_256 as OutputSizeUser>::OutputSize::USIZE;
+        let mut okm = [0u8; SHA3_OUTPUT_SIZE];
+        hk.expand(info, &mut okm)
+            .map_err(|_| signature::Error::new())?;
+
+        let secret_key = Self::PrivKey::from_bytes(ikm).map_err(|_| signature::Error::new())?;
```

Comment:
It looks like this may fail in the case that the bytes are too big (below some modulo defined for each curve).